### PR TITLE
Create an entry - Luc-Ky Khuu

### PR DESCRIPTION
**Pull request by luc-ky-khuu**
code-journal-create-entry

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -1,2 +1,40 @@
...
+
+var previousEntryArrayJSON = localStorage.getItem('code-journal-local-storage');
+if (previousEntryArrayJSON !== null) {
+  entryArray = JSON.parse(previousEntryArrayJSON);
+}
+
+window.addEventListener('beforeunload', function (events) {
+  var entryArrayJSON = JSON.stringify(entryArray);
+  localStorage.setItem('code-journal-local-storage', entryArrayJSON);
+});
```
Comment (thebearingedge) [inline js/main.js]: This should go in your `data.js`

File: js/main.js [hunk 2]
```diff
@@ -1,2 +1,40 @@
...
+
+function saveEntry() {
+  var entryObject = {};
+  entryObject.title = $form.elements.title.value;
+  entryObject.photoUrl = $form.elements.photo.value;
+  entryObject.notes = $form.elements.notes.value;
+  if (entryArray.length > 0) {
+    nextEntryId = (entryArray[0].nextEntryId + 1);
```
Comment (thebearingedge) [inline js/main.js]: Each object should get an `entryId`, not a `nextEntryId`.

File: js/main.js [hunk 3]
```diff
@@ -1,2 +1,40 @@
 /* global data */
 /* exported data */
+
+var entryArray = [];
+var nextEntryId = 0;
```
Comment (thebearingedge) [inline js/main.js]: The data model is already in `data.js`

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 1 above)
Comment (luc-ky-khuu) [reply js/main.js]: Moved code into data.js and updated variables to store entire data model and not just the user entries.

File: js/main.js (hunk 2 above)
Comment (luc-ky-khuu) [reply js/main.js]: Updated so each object gets entryId property instead of nextEntryId property

File: js/main.js (hunk 3 above)
Comment (luc-ky-khuu) [reply js/main.js]: entryArray and nextEntryId were removed from main.js

**Comment by luc-ky-khuu** [comment]
code-journal-create-entry-data-update

File: js/main.js [hunk 4]
```diff
@@ -1,40 +1,29 @@
 /* global data */
 /* exported data */
 
-var entryArray = [];
-var nextEntryId = 0;
 var $photoURL = document.querySelector('#photo');
 var $img = document.querySelector('img');
+if (data.entries.length > 1) {
+  data.nextEntryId = data.entries[0].entryId + 1;
+}
```
Comment (thebearingedge) [inline js/main.js]: This `if` on 6-9 shouldn't be necessary. What's it doing for you?

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 4 above)
Comment (luc-ky-khuu) [reply js/main.js]: if statement has been removed.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**